src/bin/tikv-ctl.rs

```diff
@@ -0,0 +1,184 @@
+                         .takes_value(true)
+                         .help("set the region id")))
+        .subcommand(SubCommand::with_name("raw")
+                    .about("print the raw node value")
+                    .version(VERSION)
+                    .author("ivan ivanyang@pingcap.com")
+                    .arg(Arg::with_name("cf")
+                         .short("c")
+                         .takes_value(true)
+                         .help("column family name, only aviable for dump-range"))
+                    .arg(Arg::with_name("key")
+                         .short("k")
+                         .takes_value(true)
+                         .help("set the query raw key, in escaped form")))
+        .subcommand(SubCommand::with_name("range")
+                    .about("print the range db range")
+                    .version(VERSION)
+                    .author("Ivan ivanyang@pingcap.com")
+                    .arg(Arg::with_name("from")
+                         .short("f")
+                         .takes_value(true)
+                         .help("set the scan from raw key, in escaped format"))
+                    .arg(Arg::with_name("to")
+                         .short("t")
+                         .takes_value(true)
+                         .help("set the scan end raw key, in escaped format"))
+                    .arg(Arg::with_name("limit")
+                         .short("l")
+                         .takes_value(true)
+                         .help("set the scan limit"))
+                    .arg(Arg::with_name("cf")
+                         .short("c")
+                         .takes_value(true)
+                         .help("column family name, only aviable for dump-range")))
+        .get_matches();
+
+    let db_path = matches.value_of("db").unwrap();
+    let db = util::rocksdb::open(db_path, ALL_CFS).unwrap();
+    if let Some(matches) = matches.subcommand_matches("raw") {
+        let cf_name = matches.value_of("cf").unwrap_or("default");
+        let key = String::from(matches.value_of("key").unwrap());
+        dump_raw_value(db, cf_name, key);
+    } else if let Some(matches) = matches.subcommand_matches("raft") {
+        let region = String::from(matches.value_of("region").unwrap());
+        let index = String::from(matches.value_of("index").unwrap());
+        dump_raft_log_entry(db, region, index);
+    } else if let Some(matches) = matches.subcommand_matches("region") {
+        let region = String::from(matches.value_of("region").unwrap());
+        dump_region_info(db, region);
+    } else if let Some(matches) = matches.subcommand_matches("range") {
+        let from = String::from(matches.value_of("from").unwrap());
+        let to = matches.value_of("to").map(|s| String::from(s));
+        let limit = matches.value_of("limit").map(|s| s.parse().unwrap());
+        let cf_name = matches.value_of("cf").unwrap_or("default");
+        dump_range(db, from, to, limit, cf_name);
```

Comment:
As a cmd tool, a sanity checking on from <= to would be helpful to provide better user experience.